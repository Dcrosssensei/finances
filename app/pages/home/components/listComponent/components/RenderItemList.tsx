import { GestureResponderEvent, StyleSheet, Text, TouchableHighlight, View } from 'react-native'
import React from 'react'
import { MaterialIcons } from '@expo/vector-icons'
import { ColorsApp } from '@/app/constants'
import { ProductTypes } from '@/app/models'

interface renderItemsProps {
    item: ProductTypes
    index: number
    selectItem: (item: ProductTypes, Index: number) => void
}

const RenderItemList = ({ item, index, selectItem }: renderItemsProps) => {
    return (
        <TouchableHighlight 
            style={styles.containerRender} 
            onPress={(event:GestureResponderEvent)=> selectItem(item, index)}
            underlayColor={ColorsApp.lightgray}
        >
            <>
                <View style={styles.elementsRender} >
                    <Text style={styles.textName}>{item.name}</Text>
                    <Text style={styles.textID}>ID: {item.id}</Text>
                </View>
                <MaterialIcons style={styles.icon} name="arrow-forward-ios" size={16} color={ColorsApp.lightgray} />
            </>

        </TouchableHighlight>
    )
}

export default RenderItemList

const styles = StyleSheet.create({
    containerRender: {
        flexDirection: 'row',
        paddingVertical: 14,
        paddingHorizontal: 10,
        borderWidth: 1,
        borderColor: "lightgray",
        alignItems: 'center'
    },
    elementsRender: {
        flex: 9
    },
    icon: {
        flex: 1
    },
    textName: {
        fontWeight: '700'
    },
    textID: {
        color: ColorsApp.gray,
        fontWeight: '500'
    }
})