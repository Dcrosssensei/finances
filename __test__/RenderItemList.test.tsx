import RenderItemList from "@/app/pages/home/components/listComponent/components/RenderItemList";
import React from "react";
import renderer from "react-test-renderer";


describe('RenderItemList', () => {
    let tree: any
    const itemTest = {
        "id": "001",
        "name": "Producto Innovador A",
        "description": "Producto A ofrece una solución integral con características avanzadas que mejoran la eficiencia y la productividad.",
        "logo": "https://handsonbanking.org/wp-content/uploads/2021/02/debitcard_front_blue.png",
        "date_release": new Date("2024-01-29"),
        "date_revision": new Date("2024-06-29")
      }

    beforeAll(()=>{
        tree = renderer.create(
        <RenderItemList 
            item={itemTest}
            index={0}
            selectItem={(item, index)=>{}}
        />
        ).toJSON();
    })
    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
      });

    it('renders name', () => {
        const nameRendered = tree.children[0].children[0].children[0];
        expect(nameRendered).toEqual("Producto Innovador A");
      });

    it('renders ID', () => {
        const nameRendered = tree.children[0].children[1].children[1];
        expect(nameRendered).toEqual("001");
      });


  });
