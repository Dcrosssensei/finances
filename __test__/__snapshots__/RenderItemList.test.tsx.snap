// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderItemList renders correctly 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "lightgray",
      "borderWidth": 1,
      "flexDirection": "row",
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      {
        "flex": 9,
      }
    }
  >
    <Text
      style={
        {
          "fontWeight": "700",
        }
      }
    >
      Producto Innovador A
    </Text>
    <Text
      style={
        {
          "color": "#636262",
          "fontWeight": "500",
        }
      }
    >
      ID: 
      001
    </Text>
  </View>
  <Text />
</View>
`;
